### PR TITLE
DS-2367 Improve BitstreamReader Performance and Caching behavior in XMLUI

### PR DESCRIPTION
**Pull request by mdiggory**
https://jira.duraspace.org/browse/DS-2367

### CODE REVIEW

File: dspace-jspui/src/main/java/org/dspace/app/webui/servlet/BitstreamServlet.java [hunk 1]
```diff
@@ -182,27 +198,78 @@ protected void doDSGet(Context context, HttpServletRequest request,
...
+
+        boolean isAnonymouslyReadable = false;
+
         if (context.getCurrentUser() == null)
         {
-            // TODO: Currently the date of the item, since we don't have dates
-            // for files
-            response.setDateHeader("Last-Modified", item.getLastModified()
-                    .getTime());
+            isAnonymouslyReadable = true;
```
Comment (mwoodiupui) [inline dspace-jspui/src/main/java/org/dspace/app/webui/servlet/BitstreamServlet.java]: I don't understand why the session being anonymous means that the bitstream is anonymously readable.

**Comment by mwoodiupui** [comment]
Overall looks reasonable.

**Comment by tdonohue** [comment]
This has merge conflicts. Also, the first commit duplicates #799 which was just closed without merging. So, commit https://github.com/atmire/DSpace/commit/8338e64bd0dc0c744693eb06152c9965f7086a6c needs removal or fixing.

This PR needs a lot of work/love before it'd be ready to get in.

**Comment by hardyoyo** [comment]
Unfortunately, this improvement PR still remains unready to merge, so we are dropping it from the 6.0 milestone. This is the second release cycle this PR has missed. I think it's going to need a dedicated sponsor to make it in for 7. Of course, the code remains available should anyone wish to incorporate it into their fork of DSpace.

**Comment by tdonohue** [comment]
While the concept behind this code is reasonable, this code is outdated/obsolete. As of the latest master, `build.properties` no longer exists (see https://jira.duraspace.org/browse/DS-2654). In additional all the included Flyway migrations would need to be renumbered (currently reference 5.0).

Closing this as outdated. Please open a new PR to continue this work.

**Pull request closed without merging**